### PR TITLE
Speed up fold batch norm

### PR DESCRIPTION
**Pull request by SmokerX**

### CODE REVIEW

**Comment by gbaned** [comment]
@petewarden  Can you please review this PR ?

**Review by petewarden — CHANGES_REQUESTED** [review]
Can you add a comment here explaining that the form used here is because the simpler approach was causing performance issues, and explain what it's doing? It only needs to be a couple of lines, thanks.

**Comment by SmokerX** [comment]
> Can you add a comment here explaining that the form used here is because the simpler approach was causing performance issues, and explain what it's doing? It only needs to be a couple of lines, thanks.

What is implemented in the original code is to process the original matrix (scaled_weights) bit by bit, let them multiply by a certain number, but because the loop is used to process the whole matrix, so the efficiency is very low, I built a sum matrix (scaled_weights) does not wait for the size of the new matrix, using numpy's own broadcast function to achieve its position-by-position multiplication, thus achieving the acceleration effect.

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Comment by gbaned** [comment]
@petewarden Can you please take a look on this PR? Thanks!

**Comment by gbaned** [comment]
@petewarden Can you please take a look on this PR? Thanks!

**Comment by mihaimaruseac** [comment]
Can you please add tests/benchmark results?

**Comment by gbaned** [comment]
@SmokerX Can you please check @mihaimaruseac's comments and keep us posted ? Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by gbaned** [comment]
@SmokerX, Any update on this PR? Please. Thanks!

**Comment by gbaned** [comment]
I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen!
